api/src/pcapi/core/offers/repository.py

```diff
@@ -550,6 +465,140 @@ def _filter_by_status(query: BaseQuery, status: str) -> BaseQuery:
     return query.filter(models.Offer.status == offer_mixin.OfferStatus[status].name)
 
 
+def _filter_collective_offers_by_statuses(query: BaseQuery, statuses: list[str] | None) -> BaseQuery:
+    """
+    Filter a SQLAlchemy query for CollectiveOffers based on a list of statuses.
+
+    This function modifies the input query to filter CollectiveOffers based on their CollectiveOfferDisplayedStatus.
+
+    Args:
+      query (BaseQuery): The initial query to be filtered.
+      statuses (list[str]): A list of status strings to filter by.
+
+    Returns:
+      BaseQuery: The modified query with applied filters.
+    """
+    status_filters: list = []
```

Comment:
En écho à mon commentaire précédent, je pense qu'il serait plus clair de nommer le filtre à partir de l'objet sur lequel il s'applique.

```suggestion
    on_collective_offer_filters: list = []
```